### PR TITLE
SSTTP-1184 - Calculator input will be reset upon sign in

### PR DESCRIPTION
**Pull request by cavster**
https://jira.tools.tax.service.gov.uk/browse/SSTTP-1184

### CODE REVIEW

File: app/uk/gov/hmrc/selfservicetimetopay/controllers/EligibilityController.scala [hunk 1]
```diff
@@ -69,13 +69,15 @@ class EligibilityController @Inject() (val messagesApi: play.api.i18n.MessagesAp
 
   def submitSignInQuestion: Action[AnyContent] = Action.async { implicit request =>
     sessionCache.get.map {
-      case Some(TTPSubmission(_, _, _, _, `validTypeOfTax`, `validExistingTTP`, cd, _, _, _)) =>
+      case Some(tpp:TTPSubmission)if tpp.eligibilityExistingTtp == `validExistingTTP` && tpp.eligibilityTypeOfTax == `validTypeOfTax`  =>
```
Comment (nkaraolis) [inline app/uk/gov/hmrc/selfservicetimetopay/controllers/EligibilityController.scala]: Rename to ttp to retain consistency throughout project

File: app/uk/gov/hmrc/selfservicetimetopay/controllers/EligibilityController.scala [hunk 2]
```diff
@@ -69,13 +69,15 @@ class EligibilityController @Inject() (val messagesApi: play.api.i18n.MessagesAp
...
         EligibilityForm.signInQuestionForm.bindFromRequest().fold(
           formWithErrors => BadRequest(sign_in_question(formWithErrors)),
           {
-            case SignInQuestion(Some(true)) => Redirect(routes.ArrangementController.determineMisalignment())
+            case SignInQuestion(Some(true)) =>
+              if(tpp.calculatorData.debits.nonEmpty) sessionCache.put(tpp.copy(calculatorData = CalculatorInput.initial))
```
Comment (nkaraolis) [inline app/uk/gov/hmrc/selfservicetimetopay/controllers/EligibilityController.scala]: Put method returns a future so might not be completed in time before redirect so probably best to flatMap on the initial future and then map on this one

**Review by nkaraolis — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/selfservicetimetopay/controllers/EligibilityController.scala (hunk 2 above)
Comment (cavster) [reply app/uk/gov/hmrc/selfservicetimetopay/controllers/EligibilityController.scala]: Hmmm take a look at submitAmountOwed in the Calculator controller then ;)

Comment (nkaraolis) [reply app/uk/gov/hmrc/selfservicetimetopay/controllers/EligibilityController.scala]: Good spot, might as well update that in this PR as well

Comment (cavster) [reply app/uk/gov/hmrc/selfservicetimetopay/controllers/EligibilityController.scala]: I'll let you have the fun of doing it ;)

**Pull request merged**